src/Entities/Character.cpp

```diff
@@ -15,4 +15,14 @@ Character::Character(){
 	sprite.setTexture(texture);
 	sprite.setColor(sf::Color(0, 255, 0));
 	this->setSprite(sprite);
+}
+
+// Set the current velocity
+void Character::setVel(sf::Vector2f vel){
+	float maxVel = 10;
+	if (vel.y>=maxVel)
+		vel.y = maxVel;
+	if(vel.x>=maxVel)
+		vel.x = maxVel;
+	vel_ = vel;
```

Comment:
velocity can be negative too. Also, this means you can move faster diagonally than you can along an axis, so it may be better to base this on speed - like scaling the magnitude of the `vel_` vector to the min of its original magnitude and a maxSpeed. Once I merge the VecUtil functions that should be easier to implement

Reply:
Yeah, I was just putting something very basic. Also not sure if all entities might need a maxVel?

Reply:
I don't think we should put it on all entities. Entity should be a very general class and who knows, some entity might need to move super fast